modules/imgproc/src/warp_kernels.simd.hpp

```diff
@@ -2130,6 +1592,1414 @@ void warpAffineLinearApproxInvoker_8UC4(const uint8_t *src_data, size_t src_step
+        int srccols = src_cols, srcrows = src_rows;
+        int dstcols = dst_cols;
+        float M[9];
+        for (int i = 0; i < 9; i++) {
+            M[i] = static_cast<float>(dM[i]);
+        }
+        uint16_t bval[] = {
+            saturate_cast<uint16_t>(border_value[0]),
+            saturate_cast<uint16_t>(border_value[1]),
+            saturate_cast<uint16_t>(border_value[2]),
+            saturate_cast<uint16_t>(border_value[3]),
+        };
+        int border_type_x = border_type != BORDER_CONSTANT &&
+                            border_type != BORDER_TRANSPARENT &&
+                            srccols <= 1 ? BORDER_REPLICATE : border_type;
+        int border_type_y = border_type != BORDER_CONSTANT &&
+                            border_type != BORDER_TRANSPARENT &&
+                            srcrows <= 1 ? BORDER_REPLICATE : border_type;
+
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+        constexpr int max_vlanes_32{VTraits<v_float32>::max_nlanes};
+        constexpr int max_uf{max_vlanes_32*2};
+        int vlanes_32 = VTraits<v_float32>::vlanes();
+        // unrolling_factor = lane_size / 16 = vlanes_32 * 32 / 16 = vlanes_32 * 2
+        int uf = vlanes_32 * 2;
+
+        std::array<float, max_vlanes_32> start_indices;
+        std::iota(start_indices.data(), start_indices.data() + max_vlanes_32, 0.f);
+
+        v_uint32 inner_srows = vx_setall_u32((unsigned)srcrows - 2),
+                 inner_scols = vx_setall_u32((unsigned)srccols - 1),
+                 outer_srows = vx_setall_u32((unsigned)srcrows + 1),
+                 outer_scols = vx_setall_u32((unsigned)srccols + 1);
+        v_float32 delta = vx_setall_f32(static_cast<float>(uf));
+        v_int32 one = vx_setall_s32(1);
+        v_int32 v_srcstep = vx_setall_s32(int(srcstep));
+        int32_t addr[max_uf],
+                src_ix[max_uf],
+                src_iy[max_uf];
+        uint16_t pixbuf[max_uf*4];
+
+        uint16_t bvalbuf[max_uf];
+        for (int i = 0; i < uf; i++) {
+            bvalbuf[i] = bval[0];
+        }
+        v_uint16 bval_v0 = vx_load(&bvalbuf[0]);
+#endif
+
+        for (int y = r.start; y < r.end; y++) {
+            uint16_t* dstptr = dst + y*dststep;
+            int x = 0;
+
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+            CV_WARPPERSPECTIVE_LINEAR_VECTOR_COMPUTE_MAPPED_COORD1();
+
+            for (; x < dstcols - uf; x += uf) {
+                // [TODO] apply halide trick
+
+                CV_WARPPERSPECTIVE_LINEAR_VECTOR_COMPUTE_MAPPED_COORD2(C1);
+
+                if (v_reduce_min(inner_mask) != 0) { // all loaded pixels are completely inside the image
+                    CV_WARP_LINEAR_VECTOR_SHUFFLE_ALLWITHIN(C1, 16U);
+                } else {
+                    CV_WARP_LINEAR_VECTOR_SHUFFLE_NOTALLWITHIN(C1, 16U);
+                }
+
+                CV_WARP_LINEAR_VECTOR_INTER_LOAD_U16(C1);
+
+                CV_WARP_LINEAR_VECTOR_INTER_LOAD_U16F32(C1);
```

Comment:
is it correct to call both LOAD_U16 and then LOAD_U16F32?

Reply:
> If the second LOAD means conversion of already loaded data from one type to another

Yes, lets rename it to `XX_CONVERT_U16F32` then.